renamed `non-indexed labels` stage to `structured metadata` (#5090)
grafana/agent: docs/sources/static/release-notes.md

* If you explicitly set the `retry_on_http_429`, no action is required.
* If you do not explicitly set `retry_on_http_429` and you do *not* want to retry on HTTP 429, make sure you set it to `false` when you upgrade to this new version.

> ### Breaking change: Renamed `non-indexed-labels` Loki processing stage to `structured_metadata`.

Reviewer: ```suggestion
### Breaking change: Renamed `non_indexed_labels` Loki processing stage to `structured_metadata`.
```

> It's necessary to renamed Loki processing stage definition `non_indexed_labels` to `structured_metadata` if this stage is used in Agent's config.

Reviewer 2: ```suggestion
If you use the Loki processing stage in your Agent configuration, you must rename the `non_indexed_labels` pipeline stage definition to `structured_metadata`.
```

We need to rephrase this sentence. Is this suggested text still technically accurate?
